.github/workflows/dotnet-build.yml

```
@@ -0,0 +1,270 @@
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@main
+
+      - name: Set Xcode version
+        if: runner.os == 'macOS'
+        run: |
+          sudo xcode-select --switch /Applications/Xcode_${{ env.CommunityToolkitSampleApp_Xcode_Version }}.app/Contents/Developer
+
+      - name: Install Latest .NET SDK
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ env.LATEST_NET_VERSION }}
+
+      - name: Install .NET MAUI Workload
+        run: dotnet workload install maui
+
+      - name: Install Tizen Workload
+        run: |
+          Invoke-WebRequest 'https://raw.githubusercontent.com/Samsung/Tizen.NET/main/workload/scripts/workload-install.ps1' -OutFile 'workload-install.ps1'
+          .\workload-install.ps1
+        shell: pwsh
+
+      - name: Display dotnet info
+        run: dotnet --info
+
+      - name: Build Community Toolkit Sample
+        run: dotnet build -c Release ${{ env.PathToCommunityToolkitSampleCsproj }}
+
+  build_library:
+    name: Build Library
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, macos-14]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@main
+
+      - name: Set NuGet Version to Tag Number
+        if: startsWith(github.ref, 'refs/tags/') && !endsWith(github.ref, '-mediaelement')
+        run: echo "NugetPackageVersion=${{ github.ref }}" >> $GITHUB_ENV
+
+      - name: Set NuGet Version to Tag Number for Camera
+        if: startsWith(github.ref, 'refs/tags/') && endsWith(github.ref, '-camera')
+        run: echo "NugetPackageVersionCamera=${{ github.ref }}" >> $GITHUB_ENV
+
+      - name: Set NuGet Version to Tag Number for MediaElement
+        if: startsWith(github.ref, 'refs/tags/') && endsWith(github.ref, '-mediaelement')
+        run: echo "NugetPackageVersionMediaElement=${{ github.ref }}" >> $GITHUB_ENV
+
+      - name: Set NuGet Version to Tag Number for Maps
+        if: startsWith(github.ref, 'refs/tags/') && endsWith(github.ref, '-maps')
+        run: echo "NugetPackageVersionMaps=${{ github.ref }}" >> $GITHUB_ENV
+
+      - name: Set NuGet Version to PR Version
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          echo "NugetPackageVersion=${{ env.CurrentSemanticVersionBase }}-build-${{ github.event.pull_request.number }}.${{ github.run_number }}+${{ github.sha }}"
+          echo "NugetPackageVersionMediaElement=${{ env.CurrentSemanticVersionBase }}-build-${{ github.event.pull_request.number }}.${{ github.run_number }}+${{ github.sha }}"
+          echo "NugetPackageVersionMaps=${{ env.CurrentSemanticVersionBase }}-build-${{ github.event.pull_request.number }}.${{ github.run_number }}+${{ github.sha }}"
+        shell: bash
+
+      - name: Set Xcode version
+        if: runner.os == 'macOS'
+        run: |
+          sudo xcode-select --switch /Applications/Xcode_${{ env.CommunityToolkitLibrary_Xcode_Version }}.app/Contents/Developer
```

We can use an existing GitHub Action for this step:

```suggestion
    - name: Install Xcode
      if: runner.os == 'macOS'
      uses: maxim-lobanov/setup-xcode@v1
      with:
        xcode-version: ${{ env.CommunityToolkitLibrary_Xcode_Version }}
```
```

---

These are one-liners and built-in macOS commands, I'd rather just stick with those than bring in a third-party dependency with all the risks associated to that.